app/grandchallenge/algorithms/migrations/0063_algorithminterface_algorithminterfacethroughtable_and_more.py

```diff
@@ -0,0 +1,137 @@
+# Generated by Django 4.2.16 on 2024-11-25 14:40
+import django.db.models.deletion
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import migrations, models
+from django.db.models import Count
+
+
+def create_algorithm_interfaces(apps, _schema_editor):
+    Algorithm = apps.get_model("algorithms", "Algorithm")  # noqa: N806
+    AlgorithmInterface = apps.get_model(  # noqa: N806
+        "algorithms", "AlgorithmInterface"
+    )
+    AlgorithmInterfaceThroughTable = apps.get_model(  # noqa: N806
+        "algorithms", "AlgorithmInterfaceThroughTable"
+    )
+    ComponentInterface = apps.get_model(  # noqa: N806
+        "components", "ComponentInterface"
+    )
+
+    for algorithm in Algorithm.objects.all():
+        inputs = ComponentInterface.objects.filter(
+            pk__in=algorithm.inputs.values_list("pk", flat=True)
+        ).all()
+        outputs = ComponentInterface.objects.filter(
+            pk__in=algorithm.outputs.values_list("pk", flat=True)
+        ).all()
```

Comment:
Why not `Algorithm.objects.prefetch_related("inputs", "outputs")` and `inputs = algorithm.inputs.all()` and `outputs = algorithm.outputs.all()`?

Reply:
I vaguely remember that this didn't work (related to the fact that migrations work with historical models), but I can't replicate it now, so your version it is.